sqlmesh/dbt/model.py

```diff
@@ -187,16 +177,21 @@ def model_kind(self, target: TargetConfig) -> ModelKind:
                     and strategy not in INCREMENTAL_BY_UNIQUE_KEY_STRATEGIES
                 ):
                     raise ConfigError(
-                        f"{self.sql_name}: SQLMesh IncrementalByUniqueKey is not compatible with '{strategy}'"
+                        f"{self.sql_name}: SQLMesh incremental by unique key strategy is not compatible with '{strategy}'"
                         f" incremental strategy. Supported strategies include {collection_to_str(INCREMENTAL_BY_UNIQUE_KEY_STRATEGIES)}."
                     )
                 return IncrementalByUniqueKeyKind(unique_key=self.unique_key, **incremental_kwargs)
 
-            raise ConfigError(
-                f"{self.sql_name}: Incremental materialization requires either a "
-                f"time_column ({collection_to_str(INCREMENTAL_BY_TIME_STRATEGIES)}) or a "
-                f"unique_key ({collection_to_str(INCREMENTAL_BY_UNIQUE_KEY_STRATEGIES.union(['none']))}) configuration."
+            logger.warning(
+                "Incremental materialization in model '%s' requires either a time_column (%s) or a unique_key (%s) configuration",
+                self.sql_name,
+                collection_to_str(INCREMENTAL_BY_TIME_STRATEGIES),
+                collection_to_str(INCREMENTAL_BY_UNIQUE_KEY_STRATEGIES.union(["none"])),
+            )
```

Comment:
The warning reads to me as something is required and I need to take action for my model to run. What do you want the user to take away from this warning? Can we be more explicit?